Library/Homebrew/formula_installer.rb

```diff
@@ -489,14 +489,16 @@ def expand_dependencies(deps)
       end
     end
 
-    if pour_bottle && !Keg.relocation_formulae.include?(formula.name)
+    if pour_bottle && !Keg.relocation_formulae.empty?
       bottle_deps = Keg.relocation_formulae
                        .map { |formula| Dependency.new(formula) }
                        .reject do |dep|
         inherited_options[dep.name] |= inherited_options_for(dep)
         dep.satisfied? inherited_options[dep.name]
       end
-      expanded_deps = Dependency.merge_repeats(bottle_deps + expanded_deps) unless bottle_deps.empty?
+      i = bottle_deps.find_index { |x| x.name == formula.name }
+      bottle_deps = bottle_deps[0, i] if i
```

Comment:
If the requested formula is one of the dependencies of `gcc`, install all the dependencies of `gcc` up to and including that formula.

Reply:
I'm not sure I understand the logic here, can you elaborate further?

Reply:
> I'm not sure I understand why this isn't done by the existing dependency logic

The only reason is that `patchelf` is not an actual dependency of anything. I have an idea of another way to approach this though.

> I'm nervous about treating the ordering of a flat list rather than dependency tree as something that can be segmented.

Given a sequence of vertices that is a topological ordering of a graph, a prefix of that sequence is also a topological order of that graph. That's not true of an arbitrary substring of a topological ordering, but is true of a prefix.

Reply:
> I'm not sure I understand why this isn't done by the existing dependency logic

The reason is that `patchelf` is an implicit dependency of everything, but not an actual dependency of anything. I have an idea of another way to approach this though.

Reply:
> The only reason is that `patchelf` is not an actual dependency of anything. I have an idea of another way to approach this though.

🆒. I think ideally it would be visible as a "proper" dependency e.g. show up in `brew deps` and co. too.

> Given a sequence of vertices that is a topological ordering of a graph, a prefix of that sequence is also a topological order of that graph. That's not true of an arbitrary substring of a topological ordering, but is true of a prefix.

😅. As long as that's covered by more naming and comment: 🆒.

Reply:
> 🆒. I think ideally it would be visible as a "proper" dependency e.g. show up in brew deps and co. too.

I'd agree, but it's a bit tricky, because it's a dependency when pouring something from a bottle, but not otherwise. We don't have that functionality in Homebrew currently.